include/io/io_mesh_ascii.tcc

```diff
@@ -525,6 +545,54 @@ std::vector<std::tuple<mpm::Index, unsigned, double>>
   return constraints;
 }
 
+//! Return acceleration constraints of nodes or particles
+template <unsigned Tdim>
+std::vector<std::tuple<mpm::Index, unsigned, double>>
+    mpm::IOMeshAscii<Tdim>::read_acceleration_constraints(
+        const std::string& acceleration_constraints_file) {
+
+  // Nodal or particle acceleration constraints
+  std::vector<std::tuple<mpm::Index, unsigned, double>> constraints;
+  constraints.clear();
+
+  // input file stream
+  std::fstream file;
+  file.open(acceleration_constraints_file.c_str(), std::ios::in);
+
+  try {
+    if (file.is_open() && file.good()) {
+      // Line
+      std::string line;
+      while (std::getline(file, line)) {
+        boost::algorithm::trim(line);
+        std::istringstream istream(line);
+        // ignore comment lines (# or !) or blank lines
+        if ((line.find('#') == std::string::npos) &&
+            (line.find('!') == std::string::npos) && (line != "")) {
+          // ID
+          mpm::Index id;
+          // Direction
+          unsigned dir;
+          // Velocity
+          double acceleration;
+          while (istream.good()) {
+            // Read stream
+            istream >> id >> dir >> acceleration;
+            constraints.emplace_back(std::make_tuple(id, dir, acceleration));
```

Comment:
Does this mean that we have to split the acceleration input into 2 in 2D and 3 in 3D? Also, the format here is .txt I guess?

Reply:
Just like velocity, displacement, or friction constraints, the acceleration will be input for each direction `(0|1|2)` individually. If you have 2D acceleration at any node, 2 inputs will be required. If you have 3D acceleration at any node, 3 inputs will be required. 

But as discussed, I think its most common for:
* 3D acceleration time history to be treated as 2D since z-dir motion is negligible
* 2D acceleration time history to be treated as 1D via RotD50, RotD100, etc. 

The applied 1D time history will require `"acceleration": 1` then rely on the math function to appropriately adjust the magnitude of accel

Reply:
Correct. Again, just like velocity, displacement, or friction constraints, if a user is inputting constant accerlations for multiple sets of nodes, they may use a `.txt` file where each row corresponds to a specific constraint for a specific node set.